x-pack/platform/plugins/shared/alerting/server/alerts_client/alerts_client.ts

```diff
@@ -157,74 +174,78 @@ export class AlertsClient<
       this.runTimestampString = opts.runTimestamp.toISOString();
     }
     await this.legacyAlertsClient.initializeExecution(opts);
+    const { executionUuid } = opts;
+
+    // No need to fetch the tracked alerts for the non-lifecycle rules
+    if (this.ruleType.autoRecoverAlerts) {
+      const getTrackedAlertsByExecutionUuid = async (uuid: string) => {
+        const result = await this.search({
```

Comment:
we need to set a size parameter, otherwise the default is 10. Maybe set it to 2 * max alerts limit? Do we need to account for anything over 10,000?

Reply:
We could potentially have max_alerts active alerts and max_alerts recovered alerts from the previous execution.

Reply:
Done